src/targeting.js

```diff
@@ -78,6 +78,25 @@ export function newTargeting(auctionManager) {
       .concat(getCustomBidTargeting(adUnitCodes, bidsReceived))
       .concat(config.getConfig('enableSendAllBids') ? getBidLandscapeTargeting(adUnitCodes, bidsReceived) : []);
 
+    // make sure at least there is a entry per adUnitCode in the targeting so receivers of SET_TARGETING call's can know what ad units are being invoked
+
+    adUnitCodes.forEach(key => {
+      if (!targeting.some(target => {
+        let hasMatchingAdUnitCode = false;
+
+        Object.getOwnPropertyNames(target).forEach(targetKey => {
+          if (targetKey === key) {
+            hasMatchingAdUnitCode = true;
+          }
+        });
+        return hasMatchingAdUnitCode;
+      })) {
+        const emptyTargeting = {};
+        emptyTargeting[key] = [];
+        targeting.push(emptyTargeting);
+      }
+    });
+
```

Comment:
I think this could be greatly simplified by replacing it with the following code _after_ the `flattenTargeting` is called below

```
    adUnitCodes.forEach(code => {
      if (!targeting[code]) {
        targeting[code] = {};
      }
    });
```

Reply:
Good point! Will give it a try